enhancements/cluster-logging/cluster-logging-log-forwarding.md

```diff
@@ -0,0 +1,320 @@
+
+## Proposal
+
+Log forwarding will provide a declarative way to specify the outputs for specific types of logs using a 'pipeline'.  A 'pipeline' defines simple routing for one source to one or more outputs. The source of logs are opinionated and well defined by cluster logging. The initial source types are as follows:
+
+* `logs.app` - Container logs generated by user applications running on the platform, excluding infrastructure container applications
+* `logs.infra` - Logs generated by both infrastructure components running on the platform and OKD nodes (e.g. journal logs).  "Infra" applications are defined as any pods which run in namespaces: `openshift*`, `kube*`, `default`.
+
+Future types may be:
+* `events.kubernetes` - Kubernetes events
+* `audit.container-engine` - Container engine audit events
+
+There are no assumptions regarding whether or not an endpoint is deployed on or off cluster.  Endpoints off-cluster may require adminstrators to perform additional actions in order for logs to be forwarded (e.g. secret creation, opening port, enable global proxy configuration) 
+Following is the list of supported endpoint types for this proposal:
+
+* Elasticsearch (v6.x)
+* Fluent secure-forward
+
+### User Stories
+
+#### As an OKD admin, I want to deploy only a collector, and have it ship logs off-cluster to my own aggregation solution
+
+This is a typical example of organizations that desires to re-use their existing enterprise log solution.  We will succeeded if we are able to write logs to their logging service.
+
+#### As an OKD admin, I want to aggregate application logs on-cluster and infra logs off-cluster
+
+This is an example of an OKD cluster hosting solution where several organizations are each provided with a dedicated cluster.  The organization requires access to application container logs but the host requires access to the infra structure logs.
+
+### Implementation Details
+
+#### Assumptions
+
+* A pipeline shall support at least one output. 
+* A pipeline may support multiple outputs.
+* Outputs that require secrets which are not created by the cluster-logging-operator shall be created and managed by the administrator of the endpoint
+* Secrets shall have keys of: `tls.crt`, `tls.key`, `ca-bundler.crt` which point to the respective certificates for which they represent. Secrets shall have the key `shared_key` for use when using secureforward 
+
+#### Security
+* A pipeline may support TLS configuration by declaring a secret. Secrets must contain keys: `tls.crt`, `tls.key`, `ca-bundler.crt`
+* Endpoints defined without a secret will use unsecure communication and must explicitly declare `insecure: true` in the output specifification
+
+#### Log Message Routing
+The `cluster-logging-operator` will create a `spec.forward` section routing both `logs.infra` and `logs.app` logs to the internal logstore if the logstore is defined in the `clusterlogging` resource and `spec.forward` is not
```

Comment:
when will it create this?  only during install/bootstrapping i'd expect.  after that i would not expect you to stomp the user's configuration, and i would expect you to just go ahead and define spec.forward.

Reply:
Added content to the upgrade section.  cc @derekwaynecarr  for awareness